pkgs/development/compilers/opensmalltalk-vm/build-vm.nix

```diff
@@ -0,0 +1,128 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchurl
+, alsa-lib
+, coreutils
+, file
+, freetype
+, gnugrep
+, libpulseaudio
+, libtool
+, libuuid
+, openssl
+, pango
+, pkg-config
+, xorg
+,
+}:
+let
+  owner = "OpenSmalltalk";
+  repo = "opensmalltalk-vm";
+  version = "202206021410";
+in
+{ platformDir, vmName, configureFlagsArray, configureFlags }:
+stdenv.mkDerivation {
+  pname =
+    let vmNameNoDots = builtins.replaceStrings [ "." ] [ "-" ] vmName;
+    in "opensmalltalk-vm-${platformDir}-${vmNameNoDots}";
+  inherit version;
+
+  src = fetchFromGitHub {
+    inherit owner repo;
+    rev = version;
+    hash = "sha256-QqElPiJuqD5svFjWrLz1zL0Tf+pHxQ2fPvkVRn2lyBI=";
+  };
+
+  postPatch =
+    let
+      inherit (builtins) substring;
+      url = "https://github.com/${owner}/${repo}.git";
+      year = substring 0 4 version;
+      month = substring 4 2 version;
+      day = substring 6 2 version;
+      hour = substring 8 2 version;
+      minute = substring 10 2 version;
+      date = "${year}-${month}-${day}T${hour}:${minute}+0000";
+      commitHash = fetchurl {
+        url = "https://api.github.com/repos/${owner}/${repo}/commits/${version}";
+        curlOpts = "--header Accept:application/vnd.github.v3.sha";
+        hash = "sha256-jhj+bngESaWC69T/fz9rMSFIvcqn7h001QMShMDrnkc=";
+      };
+      abbrevHash = substring 0 12 (builtins.readFile commitHash);
+    in
+    ''
+      vmVersionDate=$(date -u '+%a %b %-d %T %Y %z' -d "${date}")
```

Comment:
We can just hardcode this to 01.01.1970 like we do in many other places

Reply:
Since they seem to do their releases with date-based tags ([`202206021410`](https://github.com/OpenSmalltalk/opensmalltalk-vm/releases/tag/202206021410)), I feel like we should have all the metadata be consistent and extract the date from the tag.

Reply:
We don't need to do that. It is enough to just plug in the release tag but if you'd like to do it, go ahead.